### ISSUE TITLE
Access root context within other contexts (eg. range)

### ISSUE THREAD

**Issue by CheyiLin** [hairyhenderson/gomplate#719]
Hi,

Accessing root context within `range` causes error:

`tmpls/manifests.yaml`:
```yaml
useContext: {{ .v.foo.bar }}
contextInRange:
{{- range slice "a" "b" "c" }}
  - {{ .v.foo.bar }}
{{- end }}
```

`values.yaml`:
```yaml
foo:
  bar: 1024
```

results:
```
$ gomplate -V -f tmpls/manifests.yaml -c v=values.yaml
gomplate version 3.6.0, build 564f5588
config is:
input: tmpls/manifests.yaml
output: -
contexts: v=values.yaml

useContext: 1024
contextInRange:
  - rendered 0 template(s) with 1 error(s) in 272.701µs
template: tmpls/manifests.yaml:4:9: executing "tmpls/manifests.yaml" at <.v.foo.bar>: can't evaluate field v in type interface {}
```

Have found similar issue #160, just wondering is there a better way to access root context within contexts without using extra variables?

Thanks.

**Comment by hairyhenderson** [comment]
@CheyiLin Hi! Thanks for logging this.

Another option (and a better solution than using an extra variable) is to use the `$` variable, which is implicitly assigned to the root context at execution start. It's documented in the Go `text/template` docs, but it's pretty hidden away. I should also document this in the gomplate docs so it's a bit more obvious.

Anyway, this works:

`tmpls/manifests.yaml`:

```yaml
useContext: {{ .v.foo.bar }}
contextInRange:
{{- range slice "a" "b" "c" }}
  - {{ $.v.foo.bar }}
{{- end }}
```

`values.yaml`:
```yaml
foo:
  bar: 1024
```

```console:
$ gomplate -f tmpls/manifests.yaml -c v=values.yaml
useContext: 1024
contextInRange:
  - 1024
  - 1024
  - 1024
```

Hope that helps!

**Comment by hairyhenderson** [comment]
I've addressed this in #720, so I'll close this now - please let me know if you have other questions.

**Comment by CheyiLin** [comment]
@hairyhenderson Nice trick! It's much better than using extra variables. Thank you for your help!
